### PR TITLE
Add Match logic and dev route to create matches

### PR DESCRIPTION
**Pull request by manishsaha**
## Overview

Create match entity for `pear-backend`, along with repo logic and dev route to create matches.

## Changes Made

Added in the required logic for `CreateDevMatchRouter`, as well the `Match` model and `MatchRepo` logic.

## Test Coverage
Tested manually with Postman and `psql`.

## Next Steps 

Add routes required for frontend to handle matches.

### CODE REVIEW

File: src/repos/MatchRepo.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  if (users.length < 2) return false;
+  const match = db().create({
+    users,
+    status: Constants.MATCH_CREATED,
+  });
+  await db().save(match);
+  return true;
+};
+
+const getActiveMatchesByNetID = async (netID: string): Promise<Match[]> => {
```
Comment (chalo2000) [inline src/repos/MatchRepo.ts]: getWeeklyMatchesByNetID

File: src/entities/Availability.ts [hunk 2]
```diff
@@ -24,6 +25,10 @@ class Availability {
   @ManyToOne((type) => User, (user) => user.availabilities, { onDelete: 'CASCADE' })
   user: User;
 
+  /** Match with this availability */
+  @ManyToOne((type) => Match, (match) => match.availabilities, { onDelete: 'CASCADE' })
+  match: Match;
```
Comment (chalo2000) [inline src/entities/Availability.ts:30]: I thought we decided that the odds that a two matches share the same availability would be too low for the computational benefits?

File: src/utils/AppDevUtils.ts [hunk 3]
```diff
@@ -57,9 +59,21 @@ const validateURL = (url: string): boolean => {
   }
 };
 
+/* Sort matches by meetingTime in descending order */
+const sortMatchByMeetingTime = (x: Match, y: Match) => {
```
Comment (chalo2000) [inline src/utils/AppDevUtils.ts:63]: Was wondering why we put this under AppDevUtils since these are matching specific functions

**Review by chalo2000 — APPROVED** [review]
Looks good, one major question about adding matches back to availability

File: src/entities/Availability.ts (hunk 2 above)
Comment (manishsaha) [reply src/entities/Availability.ts:30]: That was from a Many to Many relationship. This Many to One relationship means many availabilities <=> one match.

File: src/utils/AppDevUtils.ts (hunk 3 above)
Comment (manishsaha) [reply src/utils/AppDevUtils.ts:63]: Functions in the repo layer are for specifically interacting with the database. Makes more sense to have standard utility functions abstracted.

File: src/repos/MatchRepo.ts (hunk 1 above)
Comment (manishsaha) [reply src/repos/MatchRepo.ts]: oo I saw the change in your PR so I avoided it, but I can update this one!

**Review by alanna-zhou — APPROVED** [review]
looks great to me :p

**Pull request merged**